internal/venice-common/src/main/java/com/linkedin/venice/pubsub/adapter/kafka/ApacheKafkaUtils.java

```diff
@@ -31,19 +38,105 @@ public static RecordHeaders convertToKafkaSpecificHeaders(PubSubMessageHeaders h
+   */
+  private static final Set<String> KAFKA_SASL_CONFIGS = Collections.unmodifiableSet(
+      new HashSet<>(
+          Arrays.asList(
+              SaslConfigs.SASL_JAAS_CONFIG,
+              SaslConfigs.SASL_MECHANISM,
+              SaslConfigs.SASL_KERBEROS_SERVICE_NAME,
+              SaslConfigs.SASL_CLIENT_CALLBACK_HANDLER_CLASS,
+              SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS,
+              SaslConfigs.SASL_LOGIN_CLASS)));
+
+  /**
+   * Extracts and returns only the valid Kafka client configuration properties from the provided
+   * {@link VeniceProperties}.
+   *
+   * <p>
+   * This method filters the provided properties against a supplied set of valid configuration
+   * keys specific to a Kafka client type (e.g., {@code ProducerConfig.configNames()},
+   * {@code ConsumerConfig.configNames()}, or {@code AdminClientConfig.configNames()}).
+   * </p>
+   *
+   * <p>
+   * In addition to the client-specific configuration keys, this method always retains common
+   * SASL-related properties defined in {@code KAFKA_SASL_CONFIGS}. If the extracted configuration
+   * specifies a Kafka security protocol that implies SSL (e.g., {@code SSL} or {@code SASL_SSL}),
+   * it also validates that all required SSL configurations are present. These required keys are
+   * defined in {@code KAFKA_SSL_MANDATORY_CONFIGS}. If any mandatory SSL property is missing or
+   * an invalid security protocol is specified, a {@link VeniceException} is thrown.
+   * </p>
+   *
+   * <p>
+   * This utility is intended for safely extracting Kafka configuration subsets suitable for
+   * initializing Kafka {@code Producer}, {@code Consumer}, or {@code AdminClient} instances.
+   * </p>
+   *
+   * @param veniceProperties The source {@link VeniceProperties} containing client configuration.
+   * @param validKafkaClientSpecificConfigKeys The set of config keys valid for the specific Kafka client type.
+   * @return A {@link Properties} object containing only valid and required Kafka client configurations.
+   * @throws VeniceException if required SSL configs are missing or an invalid protocol is specified.
+   */
+  public static Properties getValidKafkaClientProperties(
+      final VeniceProperties veniceProperties,
+      final Set<String> validKafkaClientSpecificConfigKeys) {
+    Properties extractedValidProperties = new Properties();
+
+    // Step 1: Extract properties with the specified prefixes
+    Properties strippedProperties = veniceProperties
+        .clipAndFilterNamespace(new HashSet<>(Arrays.asList(KAFKA_CONFIG_PREFIX, PUBSUB_KAFKA_CLIENT_CONFIG_PREFIX)))
+        .toProperties();
+
+    // Step 2: Filter properties based on valid Kafka client config keys
+    strippedProperties.forEach((configKey, configVal) -> {
+      if (validKafkaClientSpecificConfigKeys.contains(configKey) || KAFKA_SASL_CONFIGS.contains(configKey)) {
```

Comment:
was these SASL configs not looked at before? Seems to be newly added and extracted.

Reply:
Actually, we don’t need to explicitly check for `KAFKA_SASL_CONFIGS`, since they are already included in `validKafkaClientSpecificConfigKeys`. I’ll go ahead and remove it.

Reply:
Done!